resources/uplink_set.rb

```diff
@@ -0,0 +1,123 @@
+# (c) Copyright 2016 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+OneviewCookbook::ResourceBaseProperties.load(self)
+
+property :native_network, String
+property :networks, Array
+property :fc_networks, Array
+property :fcoe_networks, Array
+property :logical_interconnect, String
+
+default_action :create
+
+action_class do
+  include OneviewCookbook::Helper
+  include OneviewCookbook::ResourceBase
+
+  def load_native_network(item)
+    if native_network
+      if native_network.nil? || native_network == 'nil'
+        item['nativeNetworkUri'] = nil
+      else
+        native_net = OneviewSDK::EthernetNetwork.find_by(item.client, name: native_network).first
```

Comment:
Is this native network always an EthernetNetwork? The API docs weren't clear on that.